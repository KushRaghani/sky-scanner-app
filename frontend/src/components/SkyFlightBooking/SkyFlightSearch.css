.searchFlights {
  position: absolute;
  top: 30%;
  width: 80%;
  padding: 20px;
  background: rgba(255, 255, 255, 0.26);
  border-radius: 12px;
  margin: 0 10%;
  -webkit-transition: top 1s; /* For Safari 3.1 to 6.0 */
  transition: top 1s;
  z-index: 10000;
}

.searchFlights.searchClicked {
  top: 5%;
}

.searchSection {
  background: #fff;
  padding: 30px 0;
  text-align: center;
}
.flightsOption {
  width: 247px;
  height: 50px;
  background: #fff;
  margin: auto;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  line-height: 55px;
  font-size: 18px;
  font-weight: 400;
  text-align: center;
}
.selectFlights {
  display: inline-flex !important;
}
.goBtn {
  margin: 0 10px;
}
