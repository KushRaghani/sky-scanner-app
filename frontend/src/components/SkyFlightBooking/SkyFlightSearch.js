import React from "react";
import PropTypes from "prop-types";
import SkySelectAirport from "./SkySelectAirport";
import "./SkyFlightSearch.css";

const SkyFlightSearch = ({
  source,
  destination,
  onSelectChangeSource,
  onSelectChangeDest,
  getData,
  sourceAirportData,
  destinationAirportData,
  onClear,
  searchFormClasses,
  errors = {}
}) => {
  //Flight Searh component
  return (
    <React.Fragment>
      <div className={searchFormClasses}>
        <div className="flightsOption">Search Flights</div>
        <div className="searchSection">
          <div className="input-group col-lg-3 selectFlights">
            <div className="input-group-prepend">
              <label className="input-group-text">From</label>
            </div>
            <SkySelectAirport
              selectedAirport={source}
              className="custom-select"
              name="source"
              onChangeOfAirport={onSelectChangeSource}
              optionsData={sourceAirportData}
            />
            {errors && errors.source && <div>{errors.source}</div>}
          </div>
          <div className="input-group col-lg-3 selectFlights">
            <div className="input-group-prepend">
              <label className="input-group-text">To</label>
            </div>
            <SkySelectAirport
              selectedAirport={destination}
              className="custom-select"
              name="destination"
              onChangeOfAirport={onSelectChangeDest}
              optionsData={destinationAirportData}
            />
            {errors && errors.destination && <div>{errors.destination}</div>}
          </div>
          <button
            type="button"
            className="btn btn-success goBtn"
            onClick={getData}
          >
            Search
          </button>
          <button type="button" className="btn btn-danger" onClick={onClear}>
            Clear
          </button>
        </div>
      </div>
    </React.Fragment>
  );
};

SkyFlightSearch.propTypes = {
  source: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  onSelectChangeSource: PropTypes.func.isRequired,
  onSelectChangeDest: PropTypes.func.isRequired,
  getData: PropTypes.func.isRequired,
  sourceAirportData: PropTypes.arrayOf(
    PropTypes.shape({
      airportCode: PropTypes.string,
      airportName: PropTypes.string
    })
  ),
  destinationAirportData: PropTypes.arrayOf(
    PropTypes.shape({
      airportCode: PropTypes.string,
      airportName: PropTypes.string
    })
  ),
  onClear: PropTypes.func.isRequired,
  searchFormClasses: PropTypes.string.isRequired,
  errors: PropTypes.object.isRequired
};

export default SkyFlightSearch;
