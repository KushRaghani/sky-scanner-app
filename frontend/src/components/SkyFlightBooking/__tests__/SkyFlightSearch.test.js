import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SkyFlightSearch from "../SkyFlightSearch";

function loadSkyFlightSearch(args) {
  const defaultProps = {
    source: "SIN-sky",
    destination: "KUL-sky",
    onSelectChangeSource: () => {},
    onClear: () => {},
    onSelectChangeDest: () => {},
    onChangeOfAirport: () => {},
    getData: () => {},
    searchFormClasses: "",
    errors: {},
    airportData: { source: [], destination: [] },
    destinationAirportData: [],
    sourceAirportData: []
  };

  const props = { ...defaultProps, ...args };
  return <SkyFlightSearch {...props} />;
}
describe("SkyFlightSearch", () => {
  // Enyzme tests
  test("renders SkySelectAirport and button", () => {
    const wrapper = shallow(loadSkyFlightSearch());
    expect(wrapper.find("SkySelectAirport").length).toBe(2);
    expect(wrapper.find("button").length).toBe(2);
  });

  test('labels submit buttons as "Search" when not saving', () => {
    const wrapper = shallow(loadSkyFlightSearch());
    expect(
      wrapper
        .find("button")
        .at(0)
        .text()
    ).toBe("Search");
  });

  // Snapshot testing
  test("snapshot with dummy data", () => {
    const tree = renderer.create(loadSkyFlightSearch());
    expect(tree).toMatchSnapshot();
  });
});
