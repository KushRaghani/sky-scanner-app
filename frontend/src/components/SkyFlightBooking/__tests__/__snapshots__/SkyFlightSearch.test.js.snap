// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyFlightSearch snapshot with dummy data 1`] = `
<div
  className=""
>
  <div
    className="flightsOption"
  >
    Search Flights
  </div>
  <div
    className="searchSection"
  >
    <div
      className="input-group col-lg-3 selectFlights"
    >
      <div
        className="input-group-prepend"
      >
        <label
          className="input-group-text"
        >
          From
        </label>
      </div>
      <select
        className="custom-select"
        name="source"
        onChange={[Function]}
        value="SIN-sky"
      >
        <option
          value=""
        >
          Select Airport
        </option>
      </select>
    </div>
    <div
      className="input-group col-lg-3 selectFlights"
    >
      <div
        className="input-group-prepend"
      >
        <label
          className="input-group-text"
        >
          To
        </label>
      </div>
      <select
        className="custom-select"
        name="destination"
        onChange={[Function]}
        value="KUL-sky"
      >
        <option
          value=""
        >
          Select Airport
        </option>
      </select>
    </div>
    <button
      className="btn btn-success goBtn"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </div>
</div>
`;
